src/app/qgsmaptoolreverseline.cpp

```diff
@@ -176,6 +185,40 @@ QgsGeometry QgsMapToolReverseLine::partUnderPoint( QPoint point, QgsFeatureId &f
   return geomPart;
 }
 
+void QgsMapToolReverseLine::activate()
+{
+
+  QgsVectorLayer *vlayer = currentVectorLayer();
+  if ( !vlayer )
+  {
+    notifyNotVectorLayer();
+    return;
+  }
+  int nSelectedFeatures = vlayer->selectedFeatureCount();
+  if ( nSelectedFeatures > 0 )
+  {
+    QVector<bool> results;
+    QgsFeature feat;
+    QgsFeatureIterator it = vlayer->getSelectedFeatures( );
+    while ( it.nextFeature( feat ) )
+    {
+      QgsGeometry geom = feat.geometry();
+      if ( !geom.isMultipart() )
+        results.append( reverseLine( feat.id() ) );
```

Comment:
reverseLine is re-fetching the feature, so that's rather inefficient. We've already got the geometry here so you can avoid the extra feature requests by just reversing it directly